sdk/core/core-rest-pipeline/src/util/tokenCycler.ts

```diff
@@ -172,6 +173,7 @@ export function createTokenCycler(
         .then((_token) => {
           refreshWorker = null;
           token = _token;
+          tenantId = getTokenOptions.tenantId;
```

Comment:
Could there ever be other conditions under which the cycler might need to alter its mustRefresh logic? If so, would it make sense to provide a “mustRefresh” function to the cycler, and move these alterations outside of the core code?

or perhaps tenantId is the predominant reason to alter the refreshment of the cycler? if so it can stay like this for longer

Reply:
I was originally thinking about introducing a forceRefresh option, however I wasn't able to come up with a real-world scenario where that would be useful. If you can think of such scenario I'll be happy to add the forceUpdate option.

Reply:
/cc @witemple-msft for thoughts